evap/evaluation/templates/base.html

```diff
@@ -38,8 +38,8 @@
 
 {% get_current_language as LANGUAGE_CODE %}
 <div class="fixed-top d-print-none">
-    {% cache 3600 navbar request.user.email LANGUAGE_CODE %}
-        {% include_navbar user LANGUAGE_CODE %}
+    {% cache 3600 navbar request.user.email LANGUAGE_CODE in_staff_mode %}
+        {% include_navbar user LANGUAGE_CODE in_staff_mode %}
```

Comment:
Here you let the cache keep two versions, depending on whether the user is in staff mode or not, yet in `enter_staff_mode` and `exit_staff_mode` you delete the cache for both versions. Am I missing something? To me this looks like you mixed up two different approached here.

Reply:
You're right, I mixed the two different approaches. I fixed it, and now we only go with the cache deletion approach - the approach of keeping two versions in cache didn't work out due to the navbar now containing a form for the button. The cache would keep old (invalid) CSRF tokens.